test/gtest/common/mem_buffer.cc

```diff
@@ -200,31 +200,39 @@ size_t mem_buffer::get_bar1_free_size()
 {
     /* All gtest CUDA tests explicitly assume that all memory allocations are
      * done on the device 0. The same assumption is followed here. */
-    size_t available_size = SIZE_MAX;
+    static size_t available_size = SIZE_MAX;
 
 #if HAVE_CUDA
-    nvmlDevice_t device;
-    nvmlBAR1Memory_t bar1mem;
+    static bool cached_free_size = false;
```

Comment:
1. cached_free_size -> cached_available_size
2. maybe extract the code that access NVML to separate helper function?

Reply:
fixed